src/chromium/cargo.md

```diff
@@ -5,49 +5,95 @@ Chromium is built using `gn` and `ninja` and a curated set of dependencies.
-organization, and come to a group agreement about your assessment
-of risk associated with the choices above.
+Split into small groups and:
+
+* Brainstorm scenarios where `cargo` may offer an advantage
+  and assess the risk profile of these scenarios.
+* Discuss which tools, libraries, and groups of people need to be trusted
+  when using `gn` and `ninja`, offline `cargo`, etc.
 
 <details>
 
-Assuming folks taking the course are physically together, ask them to discuss
-in small groups of 3-4 people. Then, ask each table whether they've come
-to a consensus on the level of risk.
-
-Later in the course, we'll be running an actual `cargo`-based tool, `gnrt`.
-(`gnrt` uses Chromium's copy of `cargo` and `rustc`, but depends on third-party
-libraries downloaded from the internet, although `run_gnrt.py` asks `cargo` that
-only `--locked` contents are allowed via `Cargo.lock`.)
-
-Depending on the discussion, consider also mentioning the points below:
-
-* Talk about the cultural differences between the `cargo` world and the Chromium
-  world. Point out that one reason for adopting Rust in Chromium is to tap into
-  the rich, vibrant ecosystem of third-party Rust crates.
-* Talk about different risk profiles of code of the Chromium browser, Chromium
-  tests and tools, and other projects.  Point out that command like tools
-  authored in Rust are increasingly popular across the industry --- Rust tools
-  are quicker, more robust, and more ergonomic to author.
-* Talk about different trust levels that one may associate with the Rust
-  compiler, Rust tools (e.g. `rustup`, `cargo`), third-party tools (e.g. `cargo
-  audit`), and third-party libraries (e.g. crates.io libraries with different
-  popularity levels).  Point out that `rustc`, `rustup`, and `cargo` are all
-  developed under the same github organization and ask if this is a valid
-  argument for putting similar trust in all of these tools.
-* Talk about technical differences between `gn`/`ninja` vs `cargo`:
-    - Hermetic builds (e.g. using Chromium-provided `rustc` and standard library)
-    - Explicitly declaring all sources, inputs, dependencies, etc.
-    - Support for multiple languages (C++, Java, Rust, etc.) vs single language
+Ask students to avoid peeking at the speaker notes before completing the
+exercise.  Assuming folks taking the course are physically together, ask them to
+discuss in small groups of 3-4 people.
+
+Notes/hints related to the first part of the exercise ("scenarios where Cargo
+may offer an advantage"):
+
+* It's fantastic that when writing a tool, or prototyping a part of Chromium,
+  one has access to the rich ecosystem of crates.io libraries.  There is a
+  crate for almost anything and they are usually quite pleasant to use.
+  (`clap` for command-line parsing, `serde` for serializing/deserializing
+  to/from various formats, `itertools` for working with iterators, etc.).
+
+  - `cargo` makes it easy to try a library (just add a single line to
```

Comment:
I would write "Cargo" when you're talking about the tool as such, and "`cargo`" when you specifically need to refer to the binary. Here I would have used Cargo, similar with "Perl" and "Python" three lines below.

It's of course a stylistic choice and you can make your own choices for the Chromium section.

Reply:
Thanks for the feedback! I'll fix this in a follow-up PR.